README_FULL.md

```diff
@@ -0,0 +1,795 @@
+</dependency>
+<dependency>
+	<groupId>io.grpc</groupId>
+	<artifactId>grpc-netty</artifactId>
+	<version>1.27.2</version>
+</dependency>
+```
+ 
+
+### Tracer Configuration
+
+To enable tracing in your OpenTelemetry project configure a tracer bean. This bean will be autowired to controllers in your application to create and propagate spans. If you plan to use a trace exporter remember to include it in this configuration file. A sample OpenTelemetry configuration using LogExporter is shown below: 
+
+
+```java
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import io.opentelemetry.OpenTelemetry;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.export.SimpleSpansProcessor;
+import io.opentelemetry.trace.Tracer;
+
+import io.opentelemetry.exporters.logging.*;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+
+@Configuration
+@EnableAutoConfiguration(exclude = { DataSourceAutoConfiguration.class, HibernateJpaAutoConfiguration.class })
+public class OtelConfig {
+    private static final tracerName = "otel-example"	
+    @Bean
+    public Tracer otelTracer() throws Exception {
+	final Tracer tracer = OpenTelemetry.getTracerFactory().get(tracerName);
+
+	SpanProcessor logProcessor = SimpleSpansProcessor.newBuilder(new LoggingExporter()).build();
+
+	OpenTelemetrySdk.getTracerFactory().addSpanProcessor(logProcessor);
+
+	return tracer;
+    }
+}
+```
+
+
+The file above configures an OpenTelemetry tracer and a span processor which exports traces. The LoggingExporter will log spans, annotations, and events to console, providing more visibility to your application. Similarly, one could add another exporter, such as JaegerExporter, to visualize traces on different back-ends. Similar to how the LogExporter is configured, a Jaeger configuration can be added to the OtelConfig class above. 
+
+Sample configuration for a JaegerExporter:
+
+```java
+//import io.grpc.ManagedChannelBuilder;
+//import io.opentelemetry.exporters.jaeger.JaegerGrpcSpanExporter;
+
+SpanProcessor jaegerProcessor = SimpleSpansProcessor.newBuilder(JaegerGrpcSpanExporter.newBuilder()
+	.setServiceName("otel_FirstService")
+	.setChannel(ManagedChannelBuilder.forAddress("localhost", 14250).usePlaintext().build()).build())
+	.build();
+OpenTelemetrySdk.getTracerFactory().addSpanProcessor(jaegerProcessor);
+```
+     
+### Project Background
+
+Here you will create rest controllers for FirstService and SecondService.
+FirstService will send a GET request to SecondService to get the current time. FirstService will return SecondSerivce's time and it will append a message. 
+
+
+## Section 1: Manual Instrumentation with Java SDK
```

Comment:
I think this can get a little dicey. The readme is bit long and instructions can get lost in the shuffle. It might be a good idea to have clear distinctions between sections such that they're self contained. I wrote this guide so that you can follow any section without knowing about the others. I used internal links where possible to reduce duplication. Do you think this is a good approach?